GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/LooseObjectStepTests.cs

```diff
@@ -79,16 +70,74 @@ public void PutLooseObjectsInPackFiles()
         }
 
         [TestCase]
-        public void NoLooseObjectsDoesNothing()
+        [Order(3)]
+        public void RemoveLooseObjectsInPackFiles()
         {
+            // Delete/Move any starting loose objects and packfiles
             this.DeleteFiles(this.GetLooseObjectFiles());
+            this.MovePackFilesToTemp();
             this.GetLooseObjectFiles().Count.ShouldEqual(0);
-            int startingPackFileCount = this.CountPackFiles();
+            this.CountPackFiles().ShouldEqual(0);
+
+            // Copy and expand one pack
+            this.ExpandOneTempPack(copyPackBackToPackDirectory: true);
+            Assert.AreNotEqual(0, this.GetLooseObjectFiles().Count);
+            this.GetLooseObjectFiles().Count.ShouldBeAtLeast(1);
+            this.CountPackFiles().ShouldEqual(1);
 
+            // Cleanup should delete all loose objects, since they are in the packfile
             this.Enlistment.LooseObjectStep();
 
+            Assert.AreEqual(0, this.GetLooseObjectFiles().Count);
+            Assert.AreEqual(1, this.CountPackFiles());
             this.GetLooseObjectFiles().Count.ShouldEqual(0);
-            this.CountPackFiles().ShouldEqual(startingPackFileCount);
+            this.CountPackFiles().ShouldEqual(1);
+        }
+
+        [TestCase]
+        [Order(4)]
+        public void CorruptLooseObjectIsDeleted()
+        {
+            // Delete/Move any starting loose objects and packfiles
+            this.DeleteFiles(this.GetLooseObjectFiles());

```

Comment:
These 4 functions look the same as the ones in `RemoveLooseObjectsInPackFiles`.  Consider a common function?